### PR TITLE
test for PerfBalanceCallback

### LINKED ISSUE #615: x/interchainstaking/keeper: Keeper.DepositTxCallback and Keeper.PerfBalanceCallback are untested
## Summary of Bug
The code for DepositTxCallback is untested and needs comprehensive testing.

- [x] Keeper.DepositTxCallback
- [x] Keeper.PerfBalanceCallback
## Version
13bf586baffd07ca4a2a7f093212455fafb5df27

____

#### For Admin Use

- [x] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged/assigned

### PR DESCRIPTION
**Pull request by DongLieu**
## 1. Summary
Fixes #615

## 2.Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## 5. Checklist

- [ ] Does the Readme need to be updated?

### CODE REVIEW

File: x/interchainstaking/keeper/callbacks_test.go [hunk 1]
```diff
@@ -1606,6 +1606,34 @@ func (suite *KeeperTestSuite) TestDelegationAccountBalanceCallback() {
...
+		zone, _ := quicksilver.InterchainstakingKeeper.GetZone(ctx, suite.chainB.ChainID)
+		zone.DepositAddress.IncrementBalanceWaitgroup()
+		zone.WithdrawalAddress.IncrementBalanceWaitgroup()
+		quicksilver.InterchainstakingKeeper.SetZone(ctx, &zone)
+
+		response := sdk.NewCoin("qck", sdk.NewInt(10))
+		respbz, err := quicksilver.AppCodec().Marshal(&response)
+		suite.NoError(err)
+
+		for _, addr := range []string{zone.DepositAddress.Address, zone.WithdrawalAddress.Address} {
```
Comment (joe-bowman) [inline x/interchainstaking/keeper/callbacks_test.go:1627]: We should probably be passing in the performance address here. Interesting that it works regardless of what address you return a query for; but we probably should check in this fn that we are using the correct address, else the following function could still fail.

**Comment by DongLieu** [comment]
Is that okay?

**Comment by joe-bowman** [comment]
Ideally, for me, this test would not just check that nothing errors, but the outcomes of this callback:
1) has the performance account balance been updated to the value the callback passes in
2) does it trigger performance delegations to be triggered, for those validators for which they do not already exist
  - this requires setting creating performance delegations for some of the zone B valopers, and then ensuring that the correct delegation messages are sent (I don't think we have a nice way of ensuring this presently; it would be nice to hook into k.SubmitTx (or perhaps replace it in test cases), but this is a bit more work.
 
(1) should be the bare minimum in this PR; (2) is possibly a future piece, unless you want to tackle it now.

**Comment by DongLieu** [comment]
Have I done your request correctly?

**Comment by ThanhNhann** [comment]
> 2. does it trigger performance delegations to be triggered, for those validators for which they do not already exist
> 
> * this requires setting creating performance delegations for some of the zone B valopers, and then ensuring that the correct delegation messages are sent (I don't think we have a nice way of ensuring this presently; it would be nice to hook into k.SubmitTx (or perhaps replace it in test cases), but this is a bit more work.
> 
> (2) is possibly a future piece, unless you want to tackle it now.

@anhductn2001 we should make a plan for interchaintest for func like this

**Comment by ThanhNhann** [comment]
This pr linked with #615 so I updated the summary, and please fix the conflict for this @DongLieu

**Comment by DongLieu** [comment]
Ok, I fixed it @ThanhNhann

**Review by ThanhNhann — APPROVED** [review]
Lgtm! thank @DongLieu

**Review by anhductn2001 — APPROVED** [review]

**Pull request merged**